spring-xd-dirt/src/main/java/org/springframework/xd/dirt/job/dsl/Graph.java

```diff
@@ -189,10 +198,11 @@ private void followLinks(StringBuilder graphText, List<Link> toFollow, Node node
 				unvisitedNodes.remove(endOfSplit);
 				if (!endOfSplit.isSync()) {
 					// If not a sync node, include it in the output text
-					graphText.append(" || ");
+					graphText.append(" || "); 
 					printNode(graphText, endOfSplit, unvisitedNodes);
 					List<Link> transitionalLinks = findLinksFrom(endOfSplit, false);
-					printTransitions(graphText, unfollowedLinks, transitionalLinks);
+					// null final param here probably not correct
+					printTransitions(graphText, unvisitedNodes, unfollowedLinks, transitionalLinks, null);
```

Comment:
Null is not necessarily evil, but FYI: Sonar may tag us on changing the state on objects that are passed in via parameters into methods.